std/array.d

```diff
@@ -316,6 +316,66 @@ unittest
     static assert(!__traits(compiles, assocArray(b)));
 }
 
+/**
+Iterate over an associative array by key/value tuples.
```

Comment:
Technically this doesn't iterate, it just returns a range that can be lazily iterated over.